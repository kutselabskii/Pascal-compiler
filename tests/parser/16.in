program abc;

function a(b:real; d: integer): integer;
var c:integer;
begin
  b := d;
  c := 12 mod 10 + 12 div 10;	
end

var hello: real;

begin
hello := a(2.5, a(1.5, a(1, 0)));
end.