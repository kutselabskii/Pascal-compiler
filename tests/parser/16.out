|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- var
        |   |   |-- hello
        |   |   |   |__  : real
        |   |-- a	//Function_declaration
        |   |   |--  : integer
        |   |   |-- //Arguments list
        |   |   |   |-- b
        |   |   |   |   |__  : real
        |   |   |   |__ d
        |   |   |       |__  : integer
        |   |   |__ //Block
        |   |       |-- //Declaration part
        |   |       |   |-- var
        |   |       |   |   |-- c
        |   |       |   |   |   |__  : integer
        |   |       |__ //Statement part
        |   |           |__ //Statement sequence
        |   |               |-- //Assignment_statement
        |   |               |   |-- b
        |   |               |   |   |__  : real
        |   |               |   |__ //Typecast
        |   |               |       |--  : real
        |   |               |       |__ d
        |   |               |           |__  : integer
        |   |               |__ //Assignment_statement
        |   |                   |-- c
        |   |                   |   |__  : integer
        |   |                   |__ 3
        |__ //Statement part
            |__ //Statement sequence
                |-- //Assignment_statement
                |   |-- hello
                |   |   |__  : real
                |   |__ //Typecast
                |       |--  : real
                |       |__ a//Function call
                |           |--  : integer
                |           |-- 2.5
                |           |__ a//Function call
                |               |--  : integer
                |               |-- 1.5
                |               |__ a//Function call
                |                   |--  : integer
                |                   |-- //Typecast
                |                   |   |--  : real
                |                   |   |__ 1
                |                   |__ 0
